task/backend/coordinator/task_coordinator.go

```diff
@@ -0,0 +1,183 @@
+package coordinator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/task/backend"
+	"github.com/influxdata/influxdb/task/backend/executor"
+	"github.com/influxdata/influxdb/task/backend/middleware"
+	"github.com/influxdata/influxdb/task/backend/scheduler"
+	"go.uber.org/zap"
+)
+
+var _ middleware.Coordinator = (*Coordinator)(nil)
+var _ Executor = (*executor.TaskExecutor)(nil)
+var _ scheduler.SchedulableService = (*SchedulableTaskService)(nil)
+
+// Executor is an abstraction of the task executor with only the functions needed by the coordinator
+type Executor interface {
+	ManualRun(ctx context.Context, id scheduler.ID, runID scheduler.ID) (executor.Promise, error)
+	Cancel(ctx context.Context, runID scheduler.ID) error
+}
+
+type TaskCoordinator struct {
+	logger *zap.Logger
+	sch    scheduler.Scheduler
+	ex     Executor
+	// UpdateTaskFunc
+
+	limit int
+}
+
+type OptionC func(*TaskCoordinator)
+
+// SchedulableTask is a wrapper around the Task struct, giving it methods to make it compatible with the Scheduler
+type SchedulableTask struct {
+	*influxdb.Task
+}
+
+// SchedulableTaskService provides an API to update the LatestScheduled time of a task
+type SchedulableTaskService struct {
+	ts influxdb.TaskService
+}
+
+// UpdateLastScheduled uses the task service to store the latest time a task was scheduled to run
+func (c *SchedulableTaskService) UpdateLastScheduled(ctx context.Context, id scheduler.ID, t time.Time) error {
+	s := t.Format(time.RFC3339Nano)
+	tid := influxdb.ID(id)
+	_, err := c.ts.UpdateTask(ctx, tid, influxdb.TaskUpdate{
+		LatestCompleted: &s,
+	})
+
+	if err != nil {
+		return fmt.Errorf("could not update last scheduled for task; Err: %v", err)
+	}
+	return nil
+}
+
+func (t SchedulableTask) ID() scheduler.ID {
+	return scheduler.ID(t.Task.ID)
+}
+
+// Schedule takes the time a Task is scheduled for and returns a Schedule object
+func (t SchedulableTask) Schedule() scheduler.Schedule {
+	// TODO (al): use new scheduler's NewSchedule method
+	return scheduler.Schedule{}
+}
+
+// Offset returns a time.Duration for the Task's offset property
+func (t SchedulableTask) Offset() (time.Duration, error) {
```

Comment:
I think the error here should be caught before we attempt to inform the scheduler. If a task gets to the coordinator and it cant read the offset We shouldn't even attempt to communicate with the scheduler. 
What this is doing is pushing an implementation onto the scheduler. The scheduler shouldn't have to be concerned about if it can get the offset it should always have it.